From 216a2ff14c24fae27d8609b693b8405dcbe86749 Mon Sep 17 00:00:00 2001
From: Michael Zimmermann <michael.zimmermann@grandcentrix.net>
Date: Thu, 21 Jun 2018 13:22:49 +0200
Subject: [PATCH] mDNSCore: fix Warray-bounds warnings for
 HMAC_MD5_AlgName
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

../mDNSCore/DNSDigest.c: In function ‘DNSDigest_SignMessage’:
../mDNSCore/mDNSEmbeddedAPI.h:2869:32: warning: array subscript 256 is above array bounds of ‘char[26]’ [-Warray-bounds]
 #define DomainNameLength(name) DomainNameLengthLimit((name), (name)->c + MAX_DOMAIN_NAME)
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../mDNSCore/mDNSEmbeddedAPI.h:2847:57: note: in expansion of macro ‘DomainNameLength’
 #define AssignDomainName(DST, SRC) do { mDNSu16 len__ = DomainNameLength((SRC)); \
                                                         ^~~~~~~~~~~~~~~~
../mDNSCore/DNSDigest.c:1359:5: note: in expansion of macro ‘AssignDomainName’
     AssignDomainName(&tsig.resrec.rdata->u.name, &HMAC_MD5_AlgName);
     ^~~~~~~~~~~~~~~~
../mDNSCore/mDNSEmbeddedAPI.h:2869:32: warning: array subscript 256 is above array bounds of ‘char[26]’ [-Warray-bounds]
 #define DomainNameLength(name) DomainNameLengthLimit((name), (name)->c + MAX_DOMAIN_NAME)
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../mDNSCore/DNSDigest.c:1360:11: note: in expansion of macro ‘DomainNameLength’
     len = DomainNameLength(&HMAC_MD5_AlgName);
           ^~~~~~~~~~~~~~~~
---
 mDNSCore/DNSDigest.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/mDNSCore/DNSDigest.c b/mDNSCore/DNSDigest.c
index 354e2b5..6319387 100644
--- a/mDNSCore/DNSDigest.c
+++ b/mDNSCore/DNSDigest.c
@@ -1278,7 +1278,9 @@ mDNSlocal mDNSs32 DNSDigest_Base64ToBin(const char *src, mDNSu8 *target, mDNSu32
 #define HMAC_OPAD   0x5c
 #define MD5_LEN     16
 
-#define HMAC_MD5_AlgName (*(const domainname*) "\010" "hmac-md5" "\007" "sig-alg" "\003" "reg" "\003" "int")
+static const domainname HMAC_MD5_AlgName = {
+    .c = "\010" "hmac-md5" "\007" "sig-alg" "\003" "reg" "\003" "int"
+};
 
 // Adapted from Appendix, RFC 2104
 mDNSlocal void DNSDigest_ConstructHMACKey(DomainAuthInfo *info, const mDNSu8 *key, mDNSu32 len)
-- 
2.19.2

